tests/lib/rules/jsx-sort-props.js

```diff
@@ -821,5 +822,231 @@ ruleTester.run('jsx-sort-props', rule, {
+        },
+        {
+          messageId: 'sortPropsByAlpha',
+          line: 9,
+        },
+        {
+          messageId: 'sortPropsByAlpha',
+          line: 10,
+        },
+        {
+          messageId: 'sortPropsByAlpha',
+          line: 11,
+        },
+      ],
+    } : [],
+    semver.satisfies(eslintPkg.version, '> 3') ? {
+      code: `
+        <foo
+          a1={0}
+          g={0}
+          d={0} // comment for d
+          // comment for d and aa
+          aa={0}
+          c={0} // comment for c
+          // comment for c and e
+          e={1}
+          ab={1} // comment for ab
+          f={0}
+        />
+      `,
+      output: `
+        <foo
+          a1={0}
+          ab={1} // comment for ab
+          f={0}
+          g={0}
+          c={0} // comment for c
+          // comment for c and e
+          e={1}
+          d={0} // comment for d
+          // comment for d and aa
+          aa={0}
+        />
+      `,
+      errors: [
+        {
+          messageId: 'sortPropsByAlpha',
+          line: 5,
+        },
+        {
+          messageId: 'sortPropsByAlpha',
+          line: 7,
+        },
+        {
+          messageId: 'sortPropsByAlpha',
+          line: 8,
+        },
+        {
+          messageId: 'sortPropsByAlpha',
+          line: 10,
+        },
+        {
+          messageId: 'sortPropsByAlpha',
+          line: 11,
+        },
+        {
+          messageId: 'sortPropsByAlpha',
+          line: 12,
+        },
+      ],
+    } : [],
+    semver.satisfies(eslintPkg.version, '> 3') ? {
+      code: `
+        <foo
+          a1={0}
+          ab={1}
+          // comment for ab and f
+          f={0}
+          g={0}
+          c={0} // comment for c
+          // comment for c and e
+          e={1}
+          d={0}
+          aa={1} // comment for aa
+        />
+      `,
+      output: `
+        <foo
+          a1={0}
+          aa={1} // comment for aa
+          d={0}
+          g={0}
+          ab={1}
+          // comment for ab and f
+          f={0}
+          c={0} // comment for c
+          // comment for c and e
+          e={1}
+        />
+      `,
+      errors: [
+        {
+          messageId: 'sortPropsByAlpha',
+          line: 8,
+        },
+        {
+          messageId: 'sortPropsByAlpha',
+          line: 10,
+        },
+        {
+          messageId: 'sortPropsByAlpha',
+          line: 11,
+        },
+        {
+          messageId: 'sortPropsByAlpha',
+          line: 12,
+        },
+      ],
+    } : [],
+    semver.satisfies(eslintPkg.version, '> 3') ? {
+      code: `
+        <foo a={0} b={1} /* comment for b and ab */ ab={1} aa={0} />
+      `,
+      output: `
+        <foo a={0} aa={0} b={1} /* comment for b and ab */ ab={1} />
+      `,
+      errors: [
+        {
+          messageId: 'sortPropsByAlpha',
+          line: 2,
+        },
+        {
+          messageId: 'sortPropsByAlpha',
+          line: 2,
+        },
+      ],
+    } : [],
```

Comment:
```suggestion
    } : []
```
now that it's a function call, a trailing function argument comma doesn't work in the older nodes we support